src/Gaufrette/Adapter/Local.php

```diff
@@ -2,6 +2,7 @@
 
 namespace Gaufrette\Adapter;
 
+use Gaufrette\Functional\LocalDirectoryDeletor;
```

Comment:
This class is defined under the `Gaufrette\Functional` namespace, which is used for the [tests](https://github.com/KnpLabs/Gaufrette/blob/39b4e3d2ac7c50574d0f9b501342576a8236bc54/tests/Gaufrette/Functional/LocalDirectoryDeletor.php). Maybe you could move this function into the `Local` adapter itself and then use it in this class (and also update the tests to use the moved function).